lib/Mojolicious/Guides/Tutorial.pod

```diff
@@ -0,0 +1,938 @@
+    $c->render(template => 'detected');
+  };
+
+  app->start;
+  __DATA__
+
+  @@ detected.html.ep
+  <!DOCTYPE html>
+  <html>
+    <head><title>Detected</title></head>
+    <body>HTML was detected.</body>
+  </html>
+
+  @@ detected.txt.ep
+  TXT was detected.
+
+And just like with placeholders you can use a default value to make the format optional.
+
+  use Mojolicious::Lite -signatures;
+
+  # /hello
+  # /hello.json
+  # /hello.txt
+  get '/hello' => [format => ['json', 'txt']] => {format => 'txt'} => sub ($c) {
+    return $c->render(json => {hello => 'world'}) if $c->stash('format') eq 'json';
+    $c->render(text => 'hello world');
+  };
+
+  app->start;
+
+The default format is C<html> and the renderer will fall back to when necessary.
+
+=head2 Content negotiation
+
+For resources with different representations and that require truly RESTful content negotiation you can also use
+L<Mojolicious::Plugin::DefaultHelpers/"respond_to">.
+
+  use Mojolicious::Lite -signatures;
+
+  # /hello (Accept: application/json)
+  # /hello (Accept: application/xml)
+  # /hello.json
+  # /hello.xml
+  # /hello?_format=json
+  # /hello?_format=xml
+  get '/hello' => [format => ['json', 'xml']] => {format => undef} => sub ($c) {
+    $c->respond_to(
+      json => {json => {hello => 'world'}},
+      xml  => {text => '<hello>world</hello>'},
+      any  => {data => '', status => 204}
+    );
+  };
+
+  app->start;
+
+MIME type mappings can be extended or changed easily with L<Mojolicious/"types">.
+
+  app->types->type(rdf => 'application/rdf+xml');
+
+=head2 Static files
+
+Similar to templates, but with only a single file extension and optional Base64 encoding, static files can be inlined
+in the C<DATA> section and are served automatically.
+
+  use Mojolicious::Lite;
+
+  app->start;
+  __DATA__
+
+  @@ something.js
+  alert('hello!');
+
+  @@ test.txt (base64)
+  dGVzdCAxMjMKbGFsYWxh
+
+External static files are not limited to a single file extension and will be served automatically from a C<public>
+directory if it exists.
+
+  $ mkdir public
+  $ mv something.js public/something.js
+  $ mv mojolicious.tar.gz public/mojolicious.tar.gz
+
+Both have a higher precedence than routes for C<GET> and C<HEAD> requests. Content negotiation with C<Range>,
+C<If-None-Match> and C<If-Modified-Since> headers is supported as well and can be tested very easily with
+L<Mojolicious::Command::get>.
+
+  $ ./myapp.pl get /something.js -v -H 'Range: bytes=2-4'
+
+=head2 External templates
+
+External templates will be searched by the renderer in a C<templates> directory if it exists.
+
+  $ mkdir -p templates/foo
+  $ echo 'Hello World!' > templates/foo/bar.html.ep
+
+They have a higher precedence than templates in the C<DATA> section.
+
+  use Mojolicious::Lite -signatures;
+
+  # Render template "templates/foo/bar.html.ep"
+  any '/external' => sub ($c) {
+    $c->render(template => 'foo/bar');
+  };
+
+  app->start;
+
+=head2 Home
```

Comment:
A section "Log" after "Home" would be really great.  I think it would be good to have more helpful instruction about how to use Mojo::Log.  I haven't found any information for how to get hypnotoad in systemd to log to journal.  If it's not possible, should recommend to always set a log path when using hypnotoad.  Would also be nice to advise how to get logs in a log file and STDERR at the same time (I assume message event is best way).  Would also be nice to have built-in support for setting log level and path via config file.
Recommendations for how best to share the Mojo::Log object with non-controller application modules would be good too.  If the only way to do it is to pass it in object construction for each new object instance then it would be nice to advise that.